Assignment 5.Rmd

```diff
@@ -40,41 +43,60 @@ We want to see if we can build a decision tree to help teachers decide which stu
+#if score between 0.4-0.7, should be level:"teacher should monitor student progress", in short "monitor"        
+#if score greater than 0.7, should be level :"no action"   
+D1$advice <- ifelse (D1$score >= 0.4, ifelse(D1$score >= 0.7,"no action","monitor"),"intervene" )
 ```
 
 #Build a decision tree that predicts "advice" based on how many problems students have answered before, the percentage of those problems they got correct and how many hints they required
 ```{r}
-score_ctree <- 
+score_ctree <- rpart(factor(advice) ~ prior_prob_count + prior_percent_correct+hints, method="class", data=D1)
 ```
 
 #Plot tree
 ```{r}
-
+post(score_ctree, file = "score_tree.ps", title = " three groups:1:teacher should intervene, 2:teacher should monitor student progress,3:no action")
 ```
 
 Please interpret the tree, which two behaviors do you think the teacher should most closely pay attemtion to?
-
-#Test Tree
+i think the following two types of students that the teacher should most closely pay attemtion :       
+students who require hints greater than 12.5       
+students who require hints smaller than 12.5 and greater than 0.5, percentage of those problems they got correct smaller than 40%.      
+for these two types of students, they are mostly the student need to be monitored or intervened.   
+#Test Tree    
 Upload the data "intelligent_tutor_new.csv". This is a data set of a differnt sample of students doing the same problems in the same system. We can use the tree we built for the previous data set to try to predict the "advice" we should give the teacher about these new students. 
 
 ```{r}
 #Upload new data
 
-D2 <- 
+D2 <- read.csv("~/Desktop/master fall/hudk4050/assignment5/intelligent_tutor_new.csv")
 
 #Generate predicted advice using the predict() command for new students based on tree generated from old students
 
-D2$prediction <- 
-
+D2$prediction <- predict(score_ctree,D2,type = "class")
 ``` 
 ## Part III
 Compare the predicted advice with the actual advice that these students recieved. What is the difference between the observed and predicted results?
+```{r}
+library(dplyr)
+D2$advice <- ifelse (D2$score >= 0.4, ifelse(D2$score >= 0.7,'no action','monitor'),'intervene' )
+comb <- select(D2,advice,prediction)
+confMat <- table(comb)
+confMat
+accuracy <- sum(diag(confMat))/sum(confMat)
+accuracy
+```
+the accuracy is 42.5% which means the prediction only make the 42.5% corret predicts of total students. from the new data set we can see that the score of all student is 1, so all student under the level of no action, but by prediction, there are 42.5% of student should under level of no action.    
+for prediction value, the student that should be monitored is 14%, but the actual value is 0%.
+for prediction value, the student that should be intervened is 42.5%, but the actual value is 0%.
+the difference between prediction and actual value is huge.
```

Comment:
Good job! Maybe you could further comment on the acceptability of the accuracy and the model and how you would like to improve this in your future work.